packages/mock/src/msw/index.ts

```diff
@@ -103,9 +107,10 @@ export const ${handlerName} = (${isReturnHttpResponse && !isTextPlain ? `overrid
 
   return {
     implementation: {
-      function: isReturnHttpResponse
-        ? `export const ${functionName} = (${isResponseOverridable ? `overrideResponse: any = {}` : ''}): ${returnType} => (${value})\n\n`
-        : '',
+      function:
+        value && value !== 'undefined'
+          ? `export const ${functionName} = (${isResponseOverridable ? `overrideResponse: any = {}` : ''}): ${returnType} => ${isResponseOverridable ? `merge((${value}), overrideResponse)` : `(${value}) as any`}\n\n`
```

Comment:
It seems we have a type error already between the mocking result and the return type.

So far, the spread syntax for objects has resulted in an 'any' type that is broader than the intended return type, making it assignable to the return type.

This issue may handled separately.
Therefore, append ` as any ` type assertion temporarily.

FYI: https://www.typescriptlang.org/play?ts=5.2.2#code/FAFwngDgpgBAClEMC8MDeMB2BDAtrALhgGcQAnAS0wHMYBfYYAYwHtNSYA3bAGwFcoRBElRoc+IgCIQUXBEn1GrdkmyYwANV4CiasCnSLmbDjNJDEBjDAB0dvVv6wGwIA